src/components/TableList/TableList.tsx

```diff
@@ -0,0 +1,103 @@
+import { useContext } from 'react';
+import { PersonComponent } from '../PersonComponent';
+import { Person } from '../../types/Person';
+import { useSearchParams } from 'react-router-dom';
+import { SearchLink } from '../SearchLink';
+import { filterPeople } from '../../utils/filterPeople';
+import { sortPeople } from '../../utils/sortPeople';
+import { preparePeopleWithLinks } from '../../utils/preparePeopleWithLinks';
+import { TableContext } from '../../store/TableContextProvider';
+
+export const TABLE_FIELDS = ['Name', 'Sex', 'Born', 'Died', 'Mother', 'Father'];
+
+export const TableList = ({ people }: { people: Person[] }) => {
+  const { setIsEmptyMessage } = useContext(TableContext);
+  const [searchParams] = useSearchParams();
+  const sortBy = searchParams.get('sort') || '';
+  const order = searchParams.get('order') || '';
+  const centuries = searchParams.getAll('centuries') || [];
+  const query = searchParams.get('query') || '';
+  const sex = searchParams.get('sex') || '';
+
+  let preparedPeople = preparePeopleWithLinks(people);
+
+  preparedPeople = filterPeople(preparedPeople, centuries, query, sex);
+  setIsEmptyMessage(preparedPeople.length === 0);
```

Comment:
`preparedPeople.length === 0` is dynamic value, why do you set it to the strange state?

Reply:
i need isEmptyMessage in state to decide if i need to show empty message or PeopleTable. This is a bit of a strange logic, but I had to complete it at the very end. Because I missed this condition during development

Reply:
> i need isEmptyMessage in state to decide if i need to show empty message or PeopleTable. This is a bit of a strange logic, but I had to complete it at the very end. Because I missed this condition during development

It's the wrong logic, you have to show this message only if you don't have any people, but in case of state, I can manually set `isEmptyMessage` what I want.
To show the corresponding message you just need to get people from context and check this array length `people.length === 0`